File: 1446.连续字符.py
#
# @lc app=leetcode.cn id=1446 lang=python3
#
# [1446] 连续字符
#

# @lc code=start
class Solution:
    def maxPower(self, s: str) -> int:
        res = 1
        left, right = 0, 1
        while right < len(s):
            if s[right] != s[left]:
                res = max(right-left, res)
                left = right
            right += 1
        return max(right - left, res)

                
# @lc code=end

File: 1446.连续字符.cpp
/*
 * @lc app=leetcode.cn id=1446 lang=cpp
 *
 * [1446] 连续字符
 */

// @lc code=start
class Solution {
public:
    int maxPower(string s) {
        int max_char = 0;
        int start = 0;
        int pointer;
        while(start < s.size()){
            pointer = start+1;
            while(s[pointer] == s[start])
                pointer++;
            max_char = max(max_char, pointer-start);
            start = pointer;
        }
        return max(max_char, pointer-start);

    }
};
// @lc code=end

